src/App.js

```diff
@@ -17,6 +17,15 @@ const App = () => {
   const createDog = (createdDog) => {
     
   }
+
+  const updateDog = (updatedDog) => {
+    setDogs((prevDogs) =>
+      prevDogs.map((dog) => (dog.id === updatedDog.id ? updatedDog : dog))
```

Comment:
Where is prevDogs coming from?  Usually your update function in App.js should just be console logging the updatedDog until connecting with fetch.  In other words, map may be unnecessary here.